gateway/api_test.go

```diff
@@ -1554,3 +1558,80 @@ func TestRotateClientSecretHandler(t *testing.T) {
 		})
 	}
 }
+
+func TestHandleAddOrUpdateApi(t *testing.T) {
+	testFs := afero.NewMemMapFs()
+
+	t.Run("should return error when api definition json is invalid", func(t *testing.T) {
+		apiDefJson := []byte("{")
+		req, err := http.NewRequest("POST", "http://gateway", bytes.NewBuffer(apiDefJson))
+		require.NoError(t, err)
```

Comment:
I think we should remove error check for built in functions to make tests clean.

Reply:
it's not an assertion here - `require` ensures that no other side effect will break this test. This is for making tests clean.

Reply:
Nope. 
```
apiDefJson := []byte("{")
req, err := http.NewRequest("POST", "http://gateway", bytes.NewBuffer(apiDefJson))
```

This is deterministic and as a tester, you know that it will not fail because you define everything and it is a built-in function.

Reply:
See some of the built-in go tests. They are ignoring errors because no need to check each error.

Reply:
```
This is deterministic and as a tester, you know that it will not fail because you define everything and it is a built-in function.
```
Highly disagree - I cannot predict the future. So maybe the inner-working of this function will change in the future and my input may then return an error. It's a good habit to exclude any possible side-effect and I really think it's not worth it to discuss this simple line of code if a change won't make it significantly better.

Reply:
We are talking about a Go built-in function. Overthinking makes things complex instead of easier. Also, not a simple line, I saw it is repetitive.

Side note: Yes you are right but this is not a blocker to merge this one. If you don't want to change, it is ok. I just wrote my idea.